src/common/dump.cpp

```diff
@@ -230,3 +230,54 @@ int dump_hardfault_test_1(void) {
     volatile int c = 1 / b;
     return c;
 }
+
+// Dumping error message
+
+void dump_err_to_xflash(const char *error, const char *title) {
+    // This is called after w25x initialization (in dumpinfo_t dumping), so no additional w25x_init() is needed
+    uint8_t valid_flag = dump_err_in_xflash_get_valid_flag();
+    if (valid_flag != 0xFF) {                                       // ==> message space is not empty
+        if (valid_flag != 0 || dump_err_in_xflash_is_displayed()) { // ==> erase sector if flag is corrupted (!= 0x00) or is valid and is displayed
+            w25x_sector_erase(w25x_error_start_adress);
+            w25x_fetch_error();
+        }
+    }
+    dumpmessage_t dump_message;
```

Comment:
This is not initialized, later you are initializing fields `invalid`, `title` and `msg`. Field `not_displayed` is never initialized so it might contain any random data what was on stack.

Reply:
`9d28bd0`